services/transactionService.ts

```diff
@@ -798,6 +803,20 @@ export async function fetchAllPaymentsByUserIdWithPagination (
       gt: 0
     }
   }
+  if (years !== undefined && years.length > 0) {
+    const yearFilters = years.map((year) => {
+      const start = new Date(`${year}-01-01T00:00:00Z`).getTime() / 1000
```

Comment:
This same code block appears thrice in the same file, I think you could make a dedicated function for it. Also the timezone thing.